tests/unit/utils/test_serialize_dataclass.py

```diff
@@ -0,0 +1,212 @@
+        """Create a dictionary with suffixed values from a list of key-value pairs.
+
+        :param kv_pairs: The key-value pairs provided by ``dataclasses.asdict()``
+        :param suffix: The suffix to append to values
+        :returns: The dictionary with suffixed values
+        """
+        return {name: f"{str(value)}_{suffix}" for name, value in kv_pairs}
+
+
+@pytest.mark.parametrize(**ParametrizeView()._asdict())
+class Test:
+    """The dataclass serialization test class."""
+
+    @staticmethod
+    @pytest.mark.parametrize(**ParametrizeFormat()._asdict())
+    def test_content_to_dict(
+        content_view: ContentView,
+        serialization_tuple: Tuple[str, SerializationFormat],
+    ):
+        """Test the conversion of the dataclass to a dict.
+
+        :param content_view: The content view
+        :param serialization_tuple: The suffix, serialization format
+        """
+        content = ContentTestSimple()
+        result = content.asdict(
+            serialization_format=serialization_tuple[1],
+            content_view=content_view,
+        )
+        assert result == {"attr_01": False, "attr_02": 2, "attr_03": "three"}
+
+    @staticmethod
+    def test_content_to_json(content_view: ContentView):
+        """Test the conversion of the dataclass to json.
+
+        :param content_view: The content view
+        """
+        content = ContentTestSimple()
+        result = serialize(
+            content=content,
+            content_view=content_view,
+            serialization_format=SerializationFormat.JSON,
+        )
+        expected = '{\n    "attr_01": false,\n    "attr_02": 2,\n    "attr_03": "three"\n}'
+        assert result == expected
+
+    @staticmethod
+    def test_content_to_yaml(content_view: ContentView):
+        """Test the conversion of the dataclass to yaml.
+
+        :param content_view: The content view
+        """
+        content = ContentTestSimple()
+        result = serialize(
+            content=content,
+            content_view=content_view,
+            serialization_format=SerializationFormat.YAML,
+        )
+        expected = "---\nattr_01: false\nattr_02: 2\nattr_03: three\n"
+        assert result == expected
+
+    @staticmethod
+    @pytest.mark.parametrize(**ParametrizeFormat()._asdict())
+    def test_content_to_dict_override(
+        content_view: ContentView,
+        serialization_tuple: Tuple[str, SerializationFormat],
+    ):
+        """Test the conversion of the dataclass with overrides to a ``dict``.
+
+        :param content_view: The content view
+        :param serialization_tuple: The suffix, serialization format
+        """
+        content = ContentTestOverride()
+        result = content.asdict(
+            content_view=content_view,
+            serialization_format=serialization_tuple[1],
+        )
+        for _key, value in result.items():
+            assert value.endswith(f"_{serialization_tuple[0]}_{str(content_view)}")
```

Comment:
for such cases, it's customary to use `pytest-subtest` to see all of the iterations show up separately in the report.

Reply:
nice TIL, added...

Reply:
yep. stdlib unittest also has this feature, if you ever need it.